src/krb5_plugin/sssd_krb5_locator_plugin.c

```diff
@@ -581,25 +593,42 @@ krb5_error_code sssd_krb5_locator_lookup(void *private_data,
                 PLUGIN_DEBUG("getaddrinfo failed [%d][%s].\n",
                              errno, strerror(errno));
             }
+
+            if (ctx->ignore_dns_failure) {
+                continue;
+            }
+
             return KRB5_PLUGIN_NO_HANDLE;
         }
 
-        PLUGIN_DEBUG("addr[%s:%s] family[%d] socktype[%d]\n",
-                     addr[c].addr, port_str, ai->ai_family, ai->ai_socktype);
+        for (ai_item = ai; ai_item != NULL; ai_item = ai_item->ai_next) {
+            if (ctx->debug) {
+                ret = getnameinfo(ai_item->ai_addr, ai_item->ai_addrlen,
+                                  address, NI_MAXHOST,
+                                  NULL, 0,
+                                  NI_NUMERICHOST);
+                if (ret != 0) {
+                    address[0] = 0;
+                }
+            }
+
+            PLUGIN_DEBUG("addr[%s (%s)] port[%s] family[%d] socktype[%d]\n",
```

Comment:
Can you move the PLUGIN_DEBUG() inside the if (ctx->debug)-block. It is not needed but would be more readable imo.